streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java

```diff
@@ -154,26 +153,26 @@ private void runSimpleCopyTest(final int numberOfRestarts,
         output.to(outputTopic);
 
         for (int i = 0; i < numberOfRestarts; ++i) {
-            final KafkaStreams streams = new KafkaStreams(
-                builder.build(),
-                StreamsTestUtils.getStreamsConfig(
-                    applicationId,
-                    CLUSTER.bootstrapServers(),
-                    Serdes.LongSerde.class.getName(),
-                    Serdes.LongSerde.class.getName(),
-                    new Properties() {
-                        {
-                            put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, StreamsConfig.EXACTLY_ONCE);
-                            put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
-                            put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 100);
-                            put(StreamsConfig.consumerPrefix(ConsumerConfig.MAX_POLL_RECORDS_CONFIG), 1);
-                            put(StreamsConfig.consumerPrefix(ConsumerConfig.METADATA_MAX_AGE_CONFIG), "1000");
-                            put(StreamsConfig.consumerPrefix(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG), "earliest");
-                            put(IntegrationTestUtils.INTERNAL_LEAVE_GROUP_ON_CLOSE, true);
-                        }
-                    }));
-
-            try {
+            try (
+                final KafkaStreams streams = new KafkaStreams(
+                    builder.build(),
+                    StreamsTestUtils.getStreamsConfig(
+                        applicationId,
+                        CLUSTER.bootstrapServers(),
+                        Serdes.LongSerde.class.getName(),
+                        Serdes.LongSerde.class.getName(),
+                        new Properties() {
+                            {
```

Comment:
> Not your code, but double-brace initialization is not great.

Not sure why? But sure, can update.

Reply:
Maybe it's not so important specifically in the context of tests, or with Properties specifically, but in general, there are two problems stemming from the fact that this is not just creating an instance; it's actually creating an instance of an anonymous subclass.

I've included some thoughts below, but this isn't just my opinion. Searching for "double-brace initialization" brings up several discussions about it as an antipattern, not just descriptions of what it is. I'm not sure I've covered all the bases, so those other pages might be worth a read if you like.

1:
--
Anonymous subclasses create garbage in the JVM. Maybe this is better in Java 11; I'm not sure. But I bet that anonymous subclasses are still heavier than instances. At a minimum, they close over their context, so there has to be a unique class sitting in the JVM for each invocation.

2:
--
Interacting with a superclass during initialization of a subclass is not the same as interacting with an instance of the class. It's close, but subtly different, which can lead to subtle bugs. For example, inside the double braces, you can invoke protected members by accident.

There's an extra dimension to this, since the superclass can change over time. The superclass may change public members to protected, and your code would still compile even though it shouldn't.  Probably not such a concern for Properties, which hasn't changed since dinosaurs roamed the earth. But it's a reason to avoid the pattern in general.

Ideally, all classes that don't expect to be subclassed would declare themselves `final`, in which case the double-brace initialization wouldn't even work. But IMHO, it's better to be defensive and not expect authors of other classes to signal their intentions so clearly. In other words, I prefer to avoid subclassing classes unless I think they are really intended to be subclassed.

3:
--
More on the maintainability side: this pattern is a bit of an arcane invocation. There are plenty of Java developers who don't know you can do this, and plenty more who know about it but don't know why it works or what exactly it does. In other words, it's not clear if it meets the bar for simple, straightforward code that important infrastructure projects should adhere to.

For example, I'm not exactly a Java noob, but I still had to look up the initialization rules just now to make sure that the superclass's initialization blocks and constructor both get invoked before the subclass's initialization blocks.

It brings in a lot of concepts and complexity when all you really wanted to do was create a Properties inline. Why not just create and use a static method (like `mkProperties`) to do the same thing, since everyone has a very clear understanding of method invocation?